code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -905,3 +922,118 @@ butcher
+		F.hud?.update_hands() // for observers
+
+/datum/aiTask/succeedable/barricade
+	name = "barricade subtask"
+	var/has_started = FALSE
+
+/datum/aiTask/succeedable/barricade/failed()
+	var/mob/living/critter/flock/F = holder.owner
+	if(!F)
+		return TRUE
+	if(!F.can_afford(25))
+		return TRUE
+	if(get_dist(F, holder.target) > 1) //moved away before we could finish
+		return TRUE
+
+/datum/aiTask/succeedable/barricade/succeeded()
+	return is_blocked_turf(target) || (has_started && !actions.hasAction(holder.owner, "flock_construct"))
+
+/datum/aiTask/succeedable/barricade/on_tick()
+	if (!has_started && !failed() && !succeeded())
+		var/mob/living/critter/flock/drone/drone = holder.owner
+		var/dist = get_dist(drone, holder.target)
+		if(dist > 1)
+			holder.interrupt() //this should basically never happen, but sanity check just in case
+			return
+		else if(!actions.hasAction(drone, "flock_convert")) // let's not keep interrupting our own action
+			drone.set_dir(get_dir(drone, holder.target))
+			drone.hand_attack(holder.target)
+			has_started = TRUE
+
+/datum/aiTask/succeedable/barricade/on_reset()
+	has_started = FALSE
+
+/////// Targetable AI tasks, instead of looking for targets around them they just override with their own target var
+/datum/aiTask/sequence/goalbased/build/targetable
+	switched_to()
+		on_reset()
+		if (!valid_target(holder.target))
+			var/mob/living/critter/flock/drone/drone = holder.owner
+			flock_speak(drone, "Invalid conversion target provided by sentient level instruction.", drone.flock)
+			holder.interrupt()
+
+	on_reset()
+		..()
+		holder.target = get_turf(src.target)
+
+/datum/aiTask/sequence/goalbased/flockdrone_capture/targetable
+	switched_to()
+		on_reset()
+		if (!valid_target(holder.target))
+			var/mob/living/critter/flock/drone/drone = holder.owner
+			flock_speak(drone, "Invalid capture target provided by sentient level instruction.", drone.flock)
+			holder.interrupt()
+
+	on_reset()
+		..()
+		holder.target = src.target
+
+	//broader check because we want to be able to manually tell drones to capture non-enemies
+	valid_target(atom/target)
+		if (!ismob(target) && !iscritter(target) || isintangible(target))
+			return FALSE
+		if(istype(target,/mob/living))
+			var/mob/living/mob = target
+			if(!is_incapacitated(mob))
+				return FALSE
+		// mob is a valid target, check if they're not already in a cage
+		if(!istype(target.loc, /obj/flock_structure/cage))
+			return TRUE
+
+/datum/aiTask/sequence/goalbased/barricade/targetable
+	switched_to()
+		on_reset()
+		if (!valid_target(holder.target))
+			var/mob/living/critter/flock/drone/drone = holder.owner
+			flock_speak(drone, "Invalid construction target provided by sentient level instruction.", drone.flock)
+			holder.interrupt()
+
+	on_reset()
+		..()
+		holder.target = get_turf(src.target)
+
+/datum/aiTask/timed/targeted/flockdrone_shoot/targetable
+	switched_to()
+		on_reset()
+		if (!ismob(src.target) && !iscritter(src.target))
```

Comment:
```suggestion
		if (!(ismob(src.target) || iscritter(src.target) || isflock(src.target)))
```

Reply:
I think you mean `if (!(ismob(src.target) || iscritter(src.target)) || isflock(src.target))`, so I'm going to commit that